### PR TITLE
Update README to include information about S3 Versioning

### PR DESCRIPTION
**Pull request by IsaevIlya**
## Description
Updating README to include information how S3 Versioning can help with reduce checkpoints loss risks.

--------
By submitting this pull request, I confirm that my contribution is made under the terms of BSD 3-Clause License and I agree to the terms of the [LICENSE](https://github.com/awslabs/s3-connector-for-pytorch/blob/main/LICENSE).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,23 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
 usw2-az1, then the URI used will look like: `s3://my-test-bucket--usw2-az1--x-s3/<PREFIX>` (**please note that the 
 prefix for Amazon S3 Express One Zone should end with '/'**), paired with region us-west-2.
 
+## Reduce Checkpoint Loss Risks with S3 Versioning
```
Comment (muddyfish) [inline README.md:114]: I'm not sure if I like the wording of this, but I can't think of anything better off the top of my head

File: README.md [hunk 2]
```diff
@@ -111,6 +111,23 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
...
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration or multiple hosts accessing the same storage path concurrently.
```
Comment (muddyfish) [inline README.md:115]: I'd also consider advising against the async checkpointing interface, as it has race conditions

Comment (IsaevIlya) [reply README.md:115]: Nice idea.

File: README.md [hunk 3]
```diff
@@ -111,6 +111,23 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
...
+* Unintended overwrites and deletions are reversible.
+
+To enable versioning:
+1. Open the S3 console and navigate to your bucket.
+2. Click the Properties tab and Edit under Bucket Versioning.
+3. Enable Bucket Versioning and save changes.
+
+Or you can follow more detailed instructions with usage examples [Protect Data on Amazon S3 Against Accidental Deletion or Application Bugs Using S3 Versioning, S3 Object Lock, and S3 Replication](https://aws.amazon.com/getting-started/hands-on/protect-data-on-amazon-s3/?ref=docs_gateway/amazons3/DataDurability.html)
+
+With versioning enabled, your model training jobs can safely store checkpoints in S3. Any accidental modifications or deletes will be recoverable. This provides an additional layer of data durability and protection against loss.
```
[the remark this answers is not in the document]
Comment (IsaevIlya) [reply README.md:129]: Let me think about that. Not sure if it is right idea or not.

[the remark this answers is not in the document]
Comment (dnanuti) [reply README.md:129]: I agree with @muddyfish on this one, putting ourselves in the customer's shoes, we should be explicit about any cost changes that can occur. Luckily, we have an official FAQ for this we can refer to: 
https://aws.amazon.com/s3/faqs/?nc=sn&loc=7 > Q: How am I charged for using Versioning?

File: README.md (hunk 1 above)
Comment (IsaevIlya) [reply README.md]: Changed title

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (IsaevIlya) [reply README.md]: Added information about versioning costs.

File: README.md [hunk 4]
```diff
@@ -111,6 +111,18 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
...
 
+## Using S3 Versioning to Manage Checkpoints
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration, multiple hosts accessing the same storage path concurrently, or race conditions caused by usage of async checkpoint interface.
+Amazon S3 bucket versioning provides a way to protect checkpoints from inadvertent loss. With versioning enabled, S3 preserves every version of every object in the bucket. You can easily recover from unintended user actions and application failures.
+With versioning:
+* S3 stores a version history of your objects. The current version is returned by default.
+* You can retrieve previous versions by specifying the version ID.
+* Unintended overwrites and deletions are reversible.
+
+To enable versioning for S3, please follow [Enabling versioning on buckets](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html) instruction. Please keep in mind that S3 Versioning is not supported by S3 Express One Zone.
```
Comment (dnanuti) [inline README.md:122]: I'd also mention here the cost changes, something like:
> To enable versioning for S3 buckets please follow the instructions [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html). Please note that this incurs extra costs for storing object versions. For more details please check **How am I charged for using Versioning?** section in [Amazon S3 FAQs](https://aws.amazon.com/s3/faqs/?nc=sn&loc=7#Billing)

File: README.md [hunk 5]
```diff
@@ -111,6 +111,18 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
...
+
+With versioning enabled, your model training jobs can safely store checkpoints in S3. Any accidental modifications or deletes will be recoverable. This provides an additional layer of data durability and protection against loss. Please pay attention, that usage of Versioning incurs extra costs for storing objects versions. See **How am I charged for using Versioning?** section in [Amazon S3 FAQs](https://aws.amazon.com/s3/faqs/?nc=sn&loc=7#Billing).
```
Comment (dnanuti) [inline README.md:124]: > With versioning enabled, your model training jobs can safely store checkpoints in S3. Any accidental modifications or deletes will be recoverable.

Sounds redundant to what we've already said, I'd drop.

File: README.md [hunk 6]
```diff
@@ -111,6 +111,18 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
 usw2-az1, then the URI used will look like: `s3://my-test-bucket--usw2-az1--x-s3/<PREFIX>` (**please note that the 
 prefix for Amazon S3 Express One Zone should end with '/'**), paired with region us-west-2.
 
+## Using S3 Versioning to Manage Checkpoints
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration, multiple hosts accessing the same storage path concurrently, or race conditions caused by usage of async checkpoint interface.
+Amazon S3 bucket versioning provides a way to protect checkpoints from inadvertent loss. With versioning enabled, S3 preserves every version of every object in the bucket. You can easily recover from unintended user actions and application failures.
+With versioning:
```
Comment (dnanuti) [inline README.md:117]: nit: When versioning is enabled on an S3 bucket: ...

**Review by dnanuti — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -111,6 +111,18 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
 usw2-az1, then the URI used will look like: `s3://my-test-bucket--usw2-az1--x-s3/<PREFIX>` (**please note that the 
 prefix for Amazon S3 Express One Zone should end with '/'**), paired with region us-west-2.
 
+## Using S3 Versioning to Manage Checkpoints
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration, multiple hosts accessing the same storage path concurrently, or race conditions caused by usage of async checkpoint interface.
```
Comment (muddyfish) [inline README.md:115]: > by usage of async checkpoint interface.
of the async

File: README.md [hunk 8]
```diff
@@ -111,6 +111,17 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
 usw2-az1, then the URI used will look like: `s3://my-test-bucket--usw2-az1--x-s3/<PREFIX>` (**please note that the 
 prefix for Amazon S3 Express One Zone should end with '/'**), paired with region us-west-2.
 
+## Using S3 Versioning to Manage Checkpoints
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration, multiple hosts accessing the same storage path concurrently, or race conditions caused by usage of the async checkpoint interface.
+Amazon S3 bucket versioning provides a way to protect checkpoints from inadvertent loss. With versioning enabled, S3 preserves every version of every object in the bucket. You can easily recover from unintended user actions and application failures.
```
Comment (muddyfish) [inline README.md:116]: I'd remove this sentence, as I don't think it adds much
> You can easily recover from unintended user actions and application failures.

File: README.md [hunk 9]
```diff
@@ -111,6 +111,17 @@ For example, assuming the following directory bucket name `my-test-bucket--usw2-
...
+When versioning is enabled on an S3 bucket:
+* S3 stores a version history of your objects. The current version is returned by default.
+* You can retrieve previous versions by specifying the version ID.
```
Comment (muddyfish) [inline README.md:119]: I don't think "by specifying the version ID" is needed here, as the torch connector doesn't support it.

File: s3torchbenchmarking/conf/lightning_checkpointing.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+defaults:
```
Comment (muddyfish) [inline s3torchbenchmarking/conf/lightning_checkpointing.yaml:1]: Are these changes meant to be part of this PR?
